src/cargo/sources/registry/remote.rs

```diff
@@ -22,8 +22,8 @@ use std::str;
 
 fn make_crate_prefix(name: &str) -> String {
     match name.len() {
-        1 => format!("1"),
-        2 => format!("2"),
+        1 => String::from("1"),
+        2 => String::from("2"),
         3 => format!("3/{}", &name[..1]),
         _ => format!("{}/{}", &name[0..2], &name[2..4]),
```

Comment:
fwiw, I think the inconsistency here is worse.  It doesn't matter too much to me, but I feel clippy's style lints are often too pedantic.

Reply:
ya, I have looked at making this change before and decided not to for consistency.